src/main/java/org/opensearch/neuralsearch/search/query/DefaultQueryPhaseSearcherWithEmptyQueryCollectorContext.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.query;
+
+import org.apache.lucene.search.Query;
+import org.opensearch.search.internal.ContextIndexSearcher;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.query.QueryCollectorContext;
+import org.opensearch.search.query.QueryPhase;
+
+import java.io.IOException;
+import java.util.LinkedList;
+
+/**
+ * Class that inherits DefaultQueryPhaseSearcher implementation but calls its search with only
+ * empty query collector context
+ */
+public class DefaultQueryPhaseSearcherWithEmptyQueryCollectorContext extends QueryPhase.DefaultQueryPhaseSearcher {
```

Comment:
also see if we can get rid of these files and move classes inside our queryphase searcher

Reply:
ack